File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int n = nums.length;
        if(n==0)return 0;
        Arrays.sort(nums);
        int current = 1;
        int ans = 1;
        for(int i=1;i<n;i++){
            if(nums[i]==nums[i-1]+1){
                current++;
            }
            else if(nums[i]==nums[i-1]){
                continue;
            }
            else{
                current=1;
            }
            ans = Math.max(ans,current);
        }
        return ans;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = static_cast<int>(nums.size());
        vector<int> left_product(n+1,1); 
        int right_product = 1;
        vector<int> ans(n,1); 
        for(int i=0;i<n;i++){
            left_product[i+1] = left_product[i]*nums[i];
        }
        for(int i=n-1;i>0;i--){
            ans[i] = left_product[i]*right_product;
            right_product *= nums[i];
        }
        ans[0] = left_product[0]*right_product;
        return ans;
    }
};


